specs/asgi.rst

```diff
@@ -168,6 +168,11 @@ It must return another, awaitable callable::
 * ``send``, an awaitable callable taking a single event dict as a positional
   argument that will return once the send has been completed
 
+The application may have ``startup`` and ``cleanup`` attributes that
+are coroutine functions taking no arguments. If these attributes exist
+they will be awaited once per server process before serving a single
+request and after serving all requests respectively.
```

Comment:
Is there any reason why startup and cleanup couldn't be handled as message types, rather than introducing extra API? (Eg. Use something like `events.startup`, `events.cleanup`, `events.tick`)

Reply:
I don't think they could be message types received via the receive function (as that would be after serving and the scope creation) but I think they could be different scope types. Using attributes was an existing suggestion for ASGI.

Reply:
I feel a bit weird about doing it as scope types but maybe that would actually be a cleaner API, thinking about it. Would you launch one application instance for the entire server lifetime though, and send it one event at the start and one at the end?

Reply:
That’s what I’d do, yup. No new API, just a new scope type.

Reply:
OK, in that case I am +1 to that being the approach. @pgjones, what do you think?